crates/sui-checkpoint-object-store-indexer/src/main.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Context;
+use sui_checkpoint_object_store_indexer::{CheckpointBlobPipeline, EpochsPipeline};
+use sui_config::object_storage_config::ObjectStoreConfig;
+use sui_indexer_alt_framework::{Indexer, IndexerArgs};
+use sui_indexer_alt_object_store::ObjectStore;
+
+#[derive(Debug, clap::Parser)]
+#[command(name = "sui-checkpoint-object-store-indexer")]
+#[command(about = "Indexer that writes checkpoints as compressed proto blobs to object storage")]
+struct Args {
+    /// Number of concurrent checkpoint uploads
+    #[arg(long, default_value = "10")]
+    write_concurrency: usize,
+
+    /// Interval between watermark updates
+    #[arg(long, default_value = "1m", value_parser = humantime::parse_duration)]
+    watermark_interval: std::time::Duration,
+
+    #[command(flatten)]
+    object_store_config: ObjectStoreConfig,
```

Comment:
I would really like to avoid using this config -- it is strangely complex/magical for what it does, when in reality the `object_store` crate gives us most of what we want here.

I wrote something minimal on top of that crate's config types when implementing restore from formal snapshot for the consistent store -- it relies on `object_store`'s environment variable-based configuration.

https://github.com/MystenLabs/sui/blob/4d5c8b5034f08aad4bcddaa394633788dddfd8e1/crates/sui-indexer-alt-consistent-store/src/restore/formal_snapshot.rs#L28-L45

https://github.com/MystenLabs/sui/blob/4d5c8b5034f08aad4bcddaa394633788dddfd8e1/crates/sui-indexer-alt-consistent-store/src/restore/formal_snapshot.rs#L80-L111